### PR TITLE
[SRE-160] CDN redirect only for /hls

### PR DESCRIPTION
**Pull request by pwilczynskiclearcode**
This reverts commit b5a530d514d23c06b5ad64abfc5a5f29dd637fd0.

### CODE REVIEW

File: handlers/geolocation/geolocation.go
Comment (pwilczynskiclearcode) [inline handlers/geolocation/geolocation.go:143]: Compile regexp only once, during initialization.

https://pkg.go.dev/regexp#MustCompile
> MustCompile is like Compile but panics if the expression cannot be parsed. It simplifies safe initialization of global variables holding compiled regular expressions.

File: handlers/geolocation/geolocation.go
Comment (pwilczynskiclearcode) [inline handlers/geolocation/geolocation.go:144]: Compile regexp only once, during initialization.

https://pkg.go.dev/regexp#MustCompile
> MustCompile is like Compile but panics if the expression cannot be parsed. It simplifies safe initialization of global variables holding compiled regular expressions.

File: handlers/geolocation/geolocation.go
Comment (pwilczynskiclearcode) [inline handlers/geolocation/geolocation.go:42]: After deploying on staging it will be tested in the player to make sure that 406 falls back into `/hls/` endpoint.

File: balancer/balancer.go
Comment (leszko) [inline balancer/balancer.go:339]: Why do we need playback id to match this patter? 🤔

File: main.go
Comment (pwilczynskiclearcode) [inline main.go:90]: We might need to accept sth like `-cdn-redirect-playback-ids=all` to be easily press big red button for entire traffic

File: balancer/balancer.go
Comment (iameli) [reply balancer/balancer.go:339]: It's not actually a playback id if it's in this format! It's a stream key, and we match this to send them directly to a Mist endpoint instead of doing playback load balancing because it means they're streaming in 
```suggestion
var regexpStreamKey = regexp.MustCompile(`^(?:\w{4}-){3}\w{4}$`)
```

File: balancer/balancer.go
Comment (iameli) [inline balancer/balancer.go:343]:
```suggestion
	if regexpStreamKey.MatchString(playbackID) {
```

File: handlers/geolocation/geolocation.go
Comment (leszko) [reply handlers/geolocation/geolocation.go:38]: Ok, makes sense. Please document it, because it's not obvious 🙃

**Review by leszko — APPROVED** [review]
Looks good from my end, but please address the comments and wait for the final review from @iameli before merging.

File: balancer/balancer.go
Comment (pwilczynskiclearcode) [reply balancer/balancer.go:343]: That's still confusing because be pass `playbackID` into `fallbackNode()` function from `RedirectHandler()`... so in part of the code it's playbackID and now it would be considered stream-key?

File: handlers/geolocation/geolocation.go
Comment (pwilczynskiclearcode) [reply handlers/geolocation/geolocation.go:37]: Documented in the code and on the diagram https://www.notion.so/livepeer/CDN-Big-Red-Button-8e88b7f422994176a3cc8ba9773ceb16

File: main.go
Comment (iameli) [reply main.go:90]: Maybe `*` instead of `all` just so it can't collide?

**Review by iameli — APPROVED** [review]
LGTM!

**Pull request merged**
